src/endianity.rs

```diff
@@ -252,6 +255,17 @@ impl<'input, Endian> Into<&'input [u8]> for EndianBuf<'input, Endian>
     }
 }
 
+impl<'input, Endian> Read for EndianBuf<'input, Endian>
+    where Endian: Endianity
+{
+    fn read(&mut self, buf: &mut [u8]) -> io::Result<usize> {
```

Comment:
Adding `#[inline]` here made no difference. I meant moving the functions from the leb128 crate into gimli, 'inline' is probably the wrong term. We already have an impedance mismatch with the `Error` types, and now we're going to be adding a `Reader` trait that doesn't match the `io::Read` that leb128 wants. I'm not looking into performance much until this is all done, but my feeling is that if we want leb128 parsing to be fast then we'll need an implementation that more closely matches our needs.